### PR TITLE
Close Login modal when going to Privacy Policy screen

### LINKED ISSUE #1380: Close Login modal when going to Privacy Policy screen
This pop up should close when going to another screen
pi_login

### PR DESCRIPTION
**Pull request by victorgcramos**
Closes #1380

### CODE REVIEW

**Comment by victorgcramos** [comment]
Following @fernandoabolafio's suggestion, I created a new hook for route changes in order to apply the "Privacy Policy" (#1380) fix for all route redirecting.

File: src/componentsv2/Router/RouterProvider.jsx [hunk 1]
```diff
@@ -12,6 +12,27 @@ const routerCtx = createContext();
...
 
+/**
+ * Watch for route changes and triggers the callbackFn when
+ * route changes.
+ *
+ * @param {Boolean} active - if hook is active or not
+ * @param {Function} callbackFn - function that will be
+ * triggered when route changes
+ */
+export const useRouteChanges = (active, callbackFn) => {
```
Comment (fernandoabolafio) [inline src/componentsv2/Router/RouterProvider.jsx]: Please, let's move this hook to its own file under `src/hooks/utils`. And what do you think about renaming it to `useOnRouteChange`?

File: src/componentsv2/ModalLogin.jsx [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import React from "react";
 import { Modal } from "pi-ui";
 import LoginForm from "src/containers/User/Login/Form";
+import { useRouteChanges } from "src/componentsv2/Router";
 
 const ModalLogin = ({ title = "Login", onLoggedIn, onClose, ...props }) => {
+  useRouteChanges(props.show, onClose);
```
Comment (fernandoabolafio) [inline src/componentsv2/ModalLogin.jsx]: Why do you need to pass an "active" arg to this function? Couldn't it be something like:

```jsx
const ModalLogin = ({...}) => {
   useOnRouteChange(onClose);
   ...
}
```

I tested the approach above with the following hook and it worked  just fine:

```js
import { useEffect } from "react";
import { useRouter } from "src/componentsv2/Router";

export default function(callbackFn) {
  const { location } = useRouter();
  useEffect(() => {
    callbackFn();
  }, [location.pathname, callbackFn]);
}
```

File: src/componentsv2/Router/RouterProvider.jsx (hunk 1 above)
Comment (victorgcramos) [reply src/componentsv2/Router/RouterProvider.jsx]: Good idea! Working on this!

**Pull request merged**